src/compiler/parser-proc.bas

```diff
@@ -1197,12 +1228,25 @@ function cProcHeader _
 
 		case else
 			if( is_memberproc ) then
-				if( attrib and FB_SYMBATTRIB_STATIC ) then
-					errReport( FB_ERRMSG_OPERATORCANTBESTATIC, TRUE )
-					attrib and= not FB_SYMBATTRIB_STATIC
+				'' self-op? can't be static
+				if( astGetOpIsSelf( op ) ) then
+					if( attrib and FB_SYMBATTRIB_STATIC ) then
+						errReport( FB_ERRMSG_OPERATORCANTBESTATIC, TRUE )
+						attrib and= not FB_SYMBATTRIB_STATIC
+					end if
+					attrib or= FB_SYMBATTRIB_METHOD
+
+				'' must be static if a declaration
```

Comment:
Ok, non-self-ops that are members must be declared `static`. And that indirectly requires the body to be `static` too, although the `static` is optional at the body. Might be worth it to extend the comment a bit.

Reply:
Yes, definitely need to rethink the logic of `static` and `astGetOpIsSelf()`.

Need 3 types of members:
1) Self-ops, non-static only, c++ compatible name mangling
2) Non-Self-ops, non-static, c++ compatible name mangling
3) Non-Self-ops, static, fb custom name mangling

Adding the static member ops, but not the non-static member ops, will only be half of the original TODO.